### PR TITLE
Support various ip sniffers [v3]

### PR DESCRIPTION
**Pull request by luckyh**
Make the framework support various ip sniffers, and add the support of
tshark (wireshark).

The reason to do so is because tcpdump till now (the current latest
version is 4.9.2) has some problem to deal with the bootp/dhcp packet:
it will not show the 'vendor extensions/options' field when a packet
has the 'sname' or 'file' field, and that would forbid address_cache
being updated properly. So let us use tshark to workaround this issue.

---
Changes from v2 (https://github.com/avocado-framework/avocado-vt/pull/1256):

* Give warning when using tshark with ipv6 env, since it is not supported yet
* Improve accuracy of the detection about unexpected termination of the process
* Fix argument order issue of tshark

Changes from v1 (https://github.com/avocado-framework/avocado-vt/pull/1249):

* Restructure ip sniffing facilities with OOP

Signed-off-by: Xu Han xuhan@redhat.com

ID: 1516187

### CODE REVIEW

**Comment by luckyh** [comment]
Test results:

* unexpected termination
```
14:53:23 ERROR| IP sniffer (tshark) terminated unexpectedly! please check the log to get the details (status: 1)
```

* tshark ipv6 warning message
```
14:51:22 WARNI| IPv6 address sniffing is not supported yet by using TShark, please fallback to use other sniffers by uninstalling TShark when testing with IPv6
```

**Comment by luckyh** [comment]
@ldoktor could you please have a check about this version? thanks!

File: virttest/ip_sniffing.py [hunk 1]
```diff
@@ -270,5 +270,47 @@ def _output_handler(self, line):
...
+            if m_type == "5" and yiaddr != "0.0.0.0":
+                # Update cache only if get the ACK reply
+                # and the previous request is not INFORM
+                self._cache[chaddr] = yiaddr
+            return True
+
+        # DHCPv6 (RFC 3315)
+        if re.match(r"[0-9a-fA-F]{1,4}:\S+", packet[0]):
+            # TODO: support DHCPv6
+            if not self.__dict__.setdefault("_ip6_warned", False):
```
Comment (ldoktor) [inline virttest/ip_sniffing.py]: Well cleaner would be to simply have the `_ip6_warned` variable set, but this also works for me ...

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by luckyh** [comment]
OK, let's merge this, thanks.

**Pull request merged**